File: java/freeman/command/remote/RemoteControl.java
public class RemoteControl{
    Command[] onCommands;
    Command[] offCommands;
    public RemoteControl(){
        onCommands=new Command[3];
        offCommands=new Command[3];
        Command noCommand=new NoCommand();
        for (int i=0;i<3;i++){
            onCommands[i]=noCommand;
            offCommands[i]=noCommand;
        }
    }
    public void setCommand(int slot,Command onCommand,Command offCommand){
        onCommands[slot]=onCommand;
        offCommands[slot]=offCommand;
    }
    public void onButtonWasPressed(int slot){
        System.out.println("Somebody pressed on button in slot "+slot);
        onCommands[slot].execute();
    }
    public void offButtonWasPressed(int slot){
        System.out.println("Somebody pressed off button in slot "+slot);
        offCommands[slot].execute();
    }
    public String toString(){
        StringBuffer stringBuff=new StringBuffer();
        stringBuff.append("----------------Remote Control------------------\n");
        for (int i=0;i<onCommands.length;i++){
            stringBuff.append("[slot "+i+"]"+onCommands[i].getClass().getName()+" "+offCommands[i].getClass().getName()+"\n");
        }
        return stringBuff.toString();
    }
}

File: php/oop-study/lesson3/Remote/RemoteControl.php
<?php
class RemoteControl{	
	private $onCommands=array();
    private $offCommands=array();
    public function __construct(){
        $noCommand=new NoCommand();
        for ($i=0;$i<3;$i++){//Инициализируем массив пустыми командами
            $this->onCommands[i]=$noCommand;
            $this->offCommands[i]=$noCommand;
        }
    }
    public function setCommand($slot,Command $onCommand,Command $offCommand){
        $this->onCommands[$slot]=$onCommand;
        $this->offCommands[$slot]=$offCommand;
    }
    public function onButtonWasPressed($slot){
        echo "Somebody pressed on button in slot ".$slot."<br>";
        $this->onCommands[$slot]->execute();
    }
    public function offButtonWasPressed($slot){
        echo "Somebody pressed off button in slot ".$slot."<br>";
        $this->offCommands[$slot]->execute();
    }   
}
?>